plugins/network-elements/nuage-vsp/src/com/cloud/agent/api/guru/ImplementNetworkVspCommand.java

```diff
@@ -112,13 +147,212 @@ public boolean isDefaultEgressPolicy() {
+            this._networkAccountUuid = networkAccountUuid;
+            return this;
+        }
+
+        public Builder networkName(String networkName) {
+            this._networkName = networkName;
+            return this;
+        }
+
+        public Builder networkCidr(String networkCidr) {
+            this._networkCidr = networkCidr;
+            return this;
+        }
+
+        public Builder networkGateway(String networkGateway) {
+            this._networkGateway = networkGateway;
+            return this;
+        }
+
+        public Builder networkAclId(Long networkAclId) {
+            this._networkAclId = networkAclId;
+            return this;
+        }
+
+        public Builder dnsServers(List<String> dnsServers) {
+            this._dnsServers = dnsServers;
+            return this;
+        }
+
+        public Builder gatewaySystemIds(List<String> gatewaySystemIds) {
+            this._gatewaySystemIds = gatewaySystemIds;
+            return this;
+        }
+
+        public Builder networkUuid(String networkUuid) {
+            this._networkUuid = networkUuid;
+            return this;
+        }
+
+        public Builder isL3Network(boolean isL3Network) {
+            this._isL3Network = isL3Network;
+            return this;
+        }
+
+        public Builder isVpc(boolean isVpc) {
+            this._isVpc = isVpc;
+            return this;
+        }
+
+        public Builder isSharedNetwork(boolean isSharedNetwork) {
+            this._isSharedNetwork = isSharedNetwork;
+            return this;
+        }
+
+        public Builder vpcName(String vpcName) {
+            this._vpcName = vpcName;
+            return this;
+        }
+
+        public Builder vpcUuid(String vpcUuid) {
+            this._vpcUuid = vpcUuid;
+            return this;
+        }
+
+        public Builder defaultEgressPolicy(boolean defaultEgressPolicy) {
+            this._defaultEgressPolicy = defaultEgressPolicy;
+            return this;
+        }
+
+        public Builder ipAddressRange(List<String[]> ipAddressRange) {
+            this._ipAddressRange = ipAddressRange;
+            return this;
+        }
+
+        public Builder domainTemplateName(String domainTemplateName) {
+            this._domainTemplateName = domainTemplateName;
+            return this;
+        }
+
+        @Override
+        public ImplementNetworkVspCommand build() {
+            return new ImplementNetworkVspCommand(_networkDomainName, _networkDomainPath, _networkDomainUuid, _networkAccountName, _networkAccountUuid, _networkName,
+                    _networkCidr, _networkGateway, _networkAclId, _dnsServers, _gatewaySystemIds, _networkUuid, _isL3Network, _isVpc, _isSharedNetwork, _vpcName, _vpcUuid,
+                    _defaultEgressPolicy, _ipAddressRange, _domainTemplateName);
+        }
+    }
+
     @Override
     public boolean executeInSequence() {
         return false;
     }
 
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Implemented <code>equals</code> method for the <code>Command</code> super class, calling it from subclass.